src/Bicep.LangServer/Handlers/BicepCompletionHandler.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.Core;
+using Bicep.Core.SemanticModel;
+using Bicep.Core.TypeSystem;
+using Bicep.LanguageServer.CompilationManager;
+using Bicep.LanguageServer.Completions;
+using Bicep.LanguageServer.Utils;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    public class BicepCompletionHandler : CompletionHandler
+    {
+        private readonly ICompilationManager compilationManager;
+
+        public BicepCompletionHandler(ICompilationManager compilationManager)
+            : base(CreateRegistrationOptions())
+        {
+            this.compilationManager = compilationManager;
+        }
+
+        public override Task<CompletionList> Handle(CompletionParams request, CancellationToken cancellationToken)
+        {
+            var completions = GetKeywordCompletions()
+                .Concat(GetSymbolCompletions(request))
+                .Concat(GetPrimitiveTypeCompletions());
+            return Task.FromResult(new CompletionList(completions, isIncomplete: false));
+        }
+
+        public override Task<CompletionItem> Handle(CompletionItem request, CancellationToken cancellationToken)
```

Comment:
The handle and can resolve are not used by us. They are intended to delay construction of documentation. Right now, we don't need it.

Reply:
H he h

Reply:
@mymo2202 ❓